hdf5r broken on 21.11-pre

**Describe the bug**
 The hdf5r library does not work on latest release.

**To Reproduce**
`nix-shell -I 'nixpkgs=https://github.com/NixOS/nixpkgs/archive/refs/tags/21.11-pre.tar.gz' -p 'rWrapper.override { packages = [ rPackages.hdf5r ]; }'`

**Expected behavior**
 A shell with R and hdf5r.

**Additional context**
Instead I get:
```
these derivations will be built:
  /nix/store/kjdlpm9b5x29nnl0mk6kg2zwldy2nqjn-r-hdf5r-1.3.3.drv
  /nix/store/lk0rl1gsxxr32r9h5889vf6agxsfk3fy-R-4.0.4-wrapper.drv
building '/nix/store/kjdlpm9b5x29nnl0mk6kg2zwldy2nqjn-r-hdf5r-1.3.3.drv'...
unpacking sources
unpacking source archive /nix/store/fn9k1xghnb6nn8jbcll248q8njqrm79a-hdf5r_1.3.3.tar.gz
source root is hdf5r
setting SOURCE_DATE_EPOCH to timestamp 1597734147 of file hdf5r/MD5
patching sources
configuring
building
running tests
installing
* installing *source* package 'hdf5r' ...
** package 'hdf5r' successfully unpacked and MD5 sums checked
** using staged installation
checking for a sed that does not truncate output... /nix/store/s7crpcbda751bx87jyrf989ln8l6vbg3-gnused-4.8/bin/sed
checking for gawk... gawk
checking for grep that handles long lines and -e... /nix/store/0i6vphc3vnr8mg0gxjr61564hnp0s2md-gnugrep-3.6/bin/grep
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for egrep... /nix/store/0i6vphc3vnr8mg0gxjr61564hnp0s2md-gnugrep-3.6/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for h5cc... no
checking for h5pcc... no
checking for HDF5 libraries... no
configure: WARNING:
Unable to locate HDF5 compilation helper scripts 'h5cc' or 'h5pcc'.
Please specify --with-hdf5=<LOCATION> as the full path to h5cc or h5pcc.
HDF5 support is being disabled (equivalent to --with-hdf5=no).

configure: error: hdf5 does not seem to be installed on your platform.
  Please install the hdf5 library.
  The required HDF5 library files can be installed as follows:
      - Debian-based (e.g. Debian >= 8.0, Ubuntu >= 15.04): 'sudo apt-get install libhdf5-dev'
        - Old Debian-based (e.g Debian < 8.0, Ubuntu < 15.04): Install from source (see INSTALL)
        - OS X using Homebrew: 'brew install hdf5'
        - RPM-based (e.g Fedora): 'sudo yum install hdf5-devel'
ERROR: configuration failed for package 'hdf5r'
* removing '/nix/store/akjdsm72lhbd9whimnsrnbgp5gxs4p96-r-hdf5r-1.3.3/library/hdf5r'
builder for '/nix/store/kjdlpm9b5x29nnl0mk6kg2zwldy2nqjn-r-hdf5r-1.3.3.drv' failed with exit code 1
cannot build derivation '/nix/store/lk0rl1gsxxr32r9h5889vf6agxsfk3fy-R-4.0.4-wrapper.drv': 1 dependencies couldn't be built
error: build of '/nix/store/lk0rl1gsxxr32r9h5889vf6agxsfk3fy-R-4.0.4-wrapper.drv' failed
```

**Notify maintainers**
@taku0 @peti 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.0-135-generic, Ubuntu, 18.04.3 LTS (Bionic Beaver)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(ivan): `""`
 - channels(root): `"nixpkgs-21.03pre245659.30e98a4fd67"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -375,6 +375,7 @@ let
     affyio = [ pkgs.zlib.dev ];
     VariantAnnotation = [ pkgs.zlib.dev pkgs.curl.dev ];
     snpStats = [ pkgs.zlib.dev ];
+    hdf5r = [ pkgs.hdf5.dev ];
   };
 
   packagesWithBuildInputs = {

